flask/app.py

```diff
@@ -43,6 +43,8 @@
 
 def _make_timedelta(value):
     if not isinstance(value, timedelta):
+        if isinstance(value, str):
```

Comment:
you could unconditionally use `seconds=int(value)`. calling it on something that's already a number wouldn't do any harm